payu/models/staged_cable.py

```diff
@@ -152,7 +195,7 @@ def _prepare_stage(self):
             yaml.dump(self.configuration_log, conf_log_f)
 
         # Ensure the directory exists
-        if not os.path.isdir(stage_name):
+        if not os.path.isdir(os.path.join(self.control_path, stage_name)):
             errmsg = f"""Directory containing namelists for stage {stage_name}
              does not exist."""
             raise FileNotFoundError(errmsg)
```

Comment:
This might be better done earlier within/after the `__init__()` when we first read the stage configuration file. This makes sure any configuration error is picked right at the start of the process.

As it stands, if there is a directory missing for step 5, one has to run the 4 first steps to know about it, correct?

Reply:
This is true, although there could be advantages to not checking that all stage directories exist during initialisation, as it permits restarting from a later stage even if the prior stage directories no longer exist (or maybe we don't to allow this?). As it stands, one can simply ```payu sweep``` and ```payu run``` after addressing errors at a given stage, and it will just restart at the failed stage as if we had just run the prior stages. I think it makes sense to check this when we determine the stage name.